js_modules/dagster-ui/packages/ui-core/src/assets/AssetCatalogTableV2.tsx

```diff
@@ -43,103 +47,133 @@ const emptyArray: any[] = [];
-  useBlockTraceUntilTrue('useAllAssets', !!assets?.length && !assetsLoading);
+export const AssetsCatalogTableV2Impl = React.memo(
+  ({isFullScreen, toggleFullScreen}: {isFullScreen: boolean; toggleFullScreen: () => void}) => {
+    const {assets, loading: assetsLoading, error} = useAllAssets();
+    useBlockTraceUntilTrue('useAllAssets', !!assets?.length && !assetsLoading);
 
-  const [errorState, setErrorState] = useState<SyntaxError[]>([]);
-  const {filterInput, filtered, loading} = useAssetSelectionInput({
-    assets: assets ?? emptyArray,
-    assetsLoading: !assets && assetsLoading,
-    onErrorStateChange: useCallback(
-      (errors: SyntaxError[]) => {
-        if (errors !== errorState) {
-          setErrorState(errors);
-        }
-      },
-      [errorState],
-    ),
-  });
+    const [errorState, setErrorState] = useState<SyntaxError[]>([]);
+    const {filterInput, filtered, loading, setAssetSelection, assetSelection} =
+      useAssetSelectionInput<AssetTableFragment>({
+        assets: assets ?? emptyArray,
+        assetsLoading: !assets && assetsLoading,
+        onErrorStateChange: useCallback(
+          (errors: SyntaxError[]) => {
+            if (errors !== errorState) {
+              setErrorState(errors);
+            }
+          },
+          [errorState],
+        ),
+      });
 
-  const {liveDataByNode} = useAssetsHealthData(
-    useMemo(() => filtered.map((asset) => asAssetKeyInput(asset.key)), [filtered]),
-  );
+    const {liveDataByNode} = useAssetsHealthData(
+      useMemo(() => filtered.map((asset) => asAssetKeyInput(asset.key)), [filtered]),
+    );
+
+    const healthDataLoading = useMemo(() => {
+      return Object.values(liveDataByNode).length !== filtered.length;
+    }, [liveDataByNode, filtered]);
 
-  const healthDataLoading = useMemo(() => {
-    return Object.values(liveDataByNode).length !== filtered.length;
-  }, [liveDataByNode, filtered]);
+    const groupedByStatus = useMemo(() => {
+      const byStatus: Record<AssetHealthStatusString, (typeof liveDataByNode)[string][]> = {
+        Degraded: [],
+        Warning: [],
+        Healthy: [],
+        Unknown: [],
+      };
+      Object.values(liveDataByNode).forEach((asset) => {
+        const status = statusToIconAndColor[asset.assetHealth?.assetHealth ?? 'undefined'].text;
+        byStatus[status].push(asset);
+      });
+      return byStatus;
+    }, [liveDataByNode]);
 
-  const groupedByStatus = useMemo(() => {
-    const byStatus: Record<AssetHealthStatusString, (typeof liveDataByNode)[string][]> = {
-      Degraded: [],
-      Warning: [],
-      Healthy: [],
-      Unknown: [],
-    };
-    Object.values(liveDataByNode).forEach((asset) => {
-      const status = statusToIconAndColor[asset.assetHealth?.assetHealth ?? 'undefined'].text;
-      byStatus[status].push(asset);
+    const [selectedTab, setSelectedTab] = useQueryPersistedState<string>({
+      queryKey: 'selectedTab',
+      decode: (qs) => qs.selectedTab,
```

Comment:
I think dish's new qs types may require that this is `qs.selectedTab && typeof qs.selectedTab === 'string' ? qs.selectedTab : null`?